### PR TITLE
[CI] Migrate to ESRP macOS notarization

### PR DESCRIPTION
**Pull request by pjcollins**
Context: https://github.com/xamarin/release-scripts/blob/ce33e71cfe75f4ad098aa5f66c475e60582ccafb/notarize.ps1

We have received a request to migrate away from our Xamarin Notarization
logic and instead use a new ESRP Notarization script.  To do this we
need to first install the required MicroBuild plugin, and then call the
new `notarize.ps1` script.

### CODE REVIEW

**Comment by pjcollins** [comment]
@trevors20 @cadsit it looks like calling this new notarization pwsh script succeeded here - https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4283121&view=logs&j=b53b0f16-cbff-5f88-33c0-fe0867b30359&t=3c384e6e-7310-5131-4f7c-7de9943cceb5

However notarization validation failed. Does anything stand out to either of you that may be missing in how I'm calling this script?

**Comment by trevors20** [comment]
@pjcollins , I took a look.  You've uncovered a bug with the install signing plugin task where it failed but it did not fail the pipeline for some reason.  I'll log a bug for that.  But, in the meantime, it failed because it looks like the TeamName variable is not set.  Would you be able to set the TeamName variable in your pipeline and try again?

**Comment by pjcollins** [comment]
Thanks for the catch @trevors20, I just kicked off a new test build here https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4283686&view=results

**Comment by trevors20** [comment]
@pjcollins , it looks like it is failing to download the signing nuget package due to unauthorized.  Which is weird because I thought a "yaml" pipeline always had the system.accesstoken available.  But, it looks like it is undefined for some reason:  https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4283686&view=logs&j=b53b0f16-cbff-5f88-33c0-fe0867b30359&t=eca37269-4fd8-5f2b-fa6a-3864999d1b2b&l=100

On the build pipeline, it needs to allow the "OAuth" token to be used by scripts.  I know how to do that through the UI.  But, I'll try and see if I can find the yaml equivalent.  Unless you know already?

**Comment by trevors20** [comment]
@pjcollins , that's interesting.  Looks like in yaml it still needs to be defined by yourself.  From this article: https://stackoverflow.com/questions/52837980/how-to-allow-scripts-to-access-oauth-token-from-yaml-builds

It looks like there are two ways to do this.
1. pipeline level:

variables:
  system_accesstoken: $(System.AccessToken)

2. script level

- powershell: script.ps1
  env:
    SYSTEM_ACCESSTOKEN: $(System.AccessToken)

I'll update the OneNote for the yaml portion.
Since it is in a Task, would you be able to try #1 and see what happens?  The weird thing is that we are calling "tl.getVariable('System.AccessToken')" in typescript in the task.  Technically, it should have been there.  

But, in the output of the Task environment, it says it is "undefined".

**Comment by pjcollins** [comment]
Thanks for the investigation @trevors20, I think the `env:` parameter should work on any task so I've updated the PR to try that. New test run - https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=4284298&view=results

File: build-tools/automation/azure-pipelines.yaml [hunk 1]
```diff
@@ -1197,11 +1198,18 @@ stages:
...
         UnsignedPkgPath: $(XA.Unsigned.Pkg)
 
+    - task: MicroBuildSigningPlugin@3
+      displayName: Install Signing Plugin
+      inputs:
+        signType: real
+        azureSubscription: MicroBuild Signing Task (DevDiv)
+      env:
+        TeamName: XamarinAndroid
+        SYSTEM_ACCESSTOKEN: $(system.accesstoken)
```
Comment (trevors20) [inline build-tools/automation/azure-pipelines.yaml:1208]: case sensitivity for $(system.accesstoken)?  Might need to do $(System.AccessToken)

**Comment by pjcollins** [comment]
Task level variables didn't seem to work, new test job for commit dca4063: https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=4284715&view=results

**Comment by pjcollins** [comment]
It looks like the TeamName variable needs to be set on the stage/job level, and the OAUTH token needs to be passed directly to the task using the env parameter. New test run for commit fc31877: https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=4287122&view=results

**Review by pjcollins — COMMENTED** [review]
This should be good to go, the new notarization tool looks to be working as expected.

**Comment by trevors20** [comment]
I would have recommended checking this in in the early morning tomorrow so that if anything should happen people would be in the office to take a look.

**Pull request merged**